client-runtime/src/main/java/com/microsoft/rest/RestProxy.java

```diff
@@ -393,17 +396,29 @@ public HttpResponse call(String responseBody) {
                     .map(new Func1<String, Object>() {
                         @Override
                         public Object call(String responseBodyString) {
-                            return deserialize(responseBodyString, entityType, returnValueWireType);
+                            return deserialize(responseBodyString, entityType, returnValueWireType, bodyEncoding(response.headers()));
                         }
                     });
         }
         return asyncResult;
     }
 
+    private SerializerAdapter.Encoding bodyEncoding(HttpHeaders headers) {
+        String mimeContentType = headers.value("Content-Type");
+        if (mimeContentType != null) {
+            String[] parts = mimeContentType.split(";");
+            if (parts[0].equalsIgnoreCase("application/xml") || parts[0].equalsIgnoreCase("text/xml")) {
```

Comment:
Should we be checking all of the parts of the mimeContentType instead of just the first one?

Reply:
If the response content type is really a semicolon-separated list, then yes. However, it intuitively seems to me like it's not valid for there to be multiple content types on a response body, and the use of the semicolon is really to specify stuff like character encodings.